lesson25/src/js/main.js

```diff
@@ -0,0 +1,69 @@
+import { toggleModal } from "./modal";
+import * as validation from "./validation";
+
+const agreeCheckBox = document.querySelector('[data-id="agree_check"]');
+const agreeButton   = document.querySelector('[data-id="agree_button"]');
+const submitButton  = document.querySelector('[data-id="submit_button"]');
+
+agreeButton.addEventListener('click' , () => {
+    agreeCheckBox.removeAttribute('disabled');
+    agreeCheckBox.classList.remove("field-invalid");
+    agreeCheckBox.checked = true;
+    changeDisabledStatusSubmitButton();
+});
+
+submitButton.addEventListener("click", (e) => {
+    e.preventDefault();
+    window.location.href = "register-done.html";
+});
+
+const options = {
+    root: document.querySelector('[data-id="modal-inner"]'),
+    threshold: 1
+};
+
+const removeDisabledForAgreeButton = ([entry]) => {
+    if(entry.isIntersecting){
+        const agreeButton  = document.querySelector('[data-id="agree_button"]');
```

Comment:
IMO
Since the agreeButton is already defined globally, you can remove it :)

Reply:
Thank you for your comment!
I fixed.
https://github.com/nakagawamai/js-lesson/pull/27/commits/04717621cd73cdede168494d24d4d3a306198cbf